pkg/stamper/stamper.go

```diff
@@ -0,0 +1,269 @@
+package stamper
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"time"
+
+	"github.com/ethersphere/beekeeper/pkg/bee"
+	"github.com/ethersphere/beekeeper/pkg/bee/api"
+	"github.com/ethersphere/beekeeper/pkg/k8s"
+	"github.com/ethersphere/beekeeper/pkg/logging"
+	"github.com/ethersphere/beekeeper/pkg/swap"
+)
+
+// Client is the interface for stamper client.
+type Client interface {
+	// Create creates a postage batch.
+	Create(ctx context.Context, amount uint64, depth uint16) error
+	// Dilute dilutes a postage batch.
+	Dilute(ctx context.Context, threshold float64, depth uint16, opts ...Option) error
+	// Set sets the topup and dilution parameters.
+	Set(ctx context.Context, ttlThreshold, topupDuration time.Duration, threshold float64, depth uint16, opts ...Option) error
+	// Topup tops up a postage batch.
+	Topup(ctx context.Context, ttlThreshold, topupDuration time.Duration, opts ...Option) error
+}
+
+type Option func(*options)
+
+type options struct {
+	batchIds []string
+}
+
+func WithBatchIDs(batchIds []string) Option {
+	return func(o *options) {
+		o.batchIds = batchIds
+	}
+}
+
+type ClientConfig struct {
+	Log           logging.Logger
+	Namespace     string
+	K8sClient     *k8s.Client
+	SwapClient    swap.Client
+	BeeClients    map[string]*bee.Client
+	LabelSelector string
+	InCluster     bool
+}
+
+type client struct {
+	log           logging.Logger
+	namespace     string
+	k8sClient     *k8s.Client
+	swapClient    swap.Client
+	beeClients    map[string]*bee.Client
+	labelSelector string
+	inCluster     bool
+}
+
+func New(cfg *ClientConfig) *client {
```

Comment:
As `client` is not exported, maybe it is nicer to return the `Client` interface in this function signature.

Reply:
I have exporte the Client and removed the interface.

Reply:
Great.